### PR TITLE
Added a parser

### PR DESCRIPTION
**Pull request by weshooper**
Added a parser to provide a simple interface for retrieving assignments/categories/products from Demandware XML files.

Currently only returns select elements/details (to expand later as/when needed), along with all custom attributes.

Have also skipped validation of XML files against XSD schemas for files over 500MB, since at that size, DomDocument silently exits the script. Needs looking at another time, but don’t typically see files that large.

### CODE REVIEW

File: src/Parser.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
+<?php
+namespace DemandwareXml;
+
+use \Exception;
```
Comment (ziadoz) [inline src/Parser.php]: Either throw the `Demandware\XmlException` or maybe add a new `Demandware\ParserException`?

Comment (weshooper) [inline src/Parser.php]: @ziadoz good spot, done!

File: tests/ParserTest.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        $expected = json_decode(file_get_contents($path . 'categories.json'), true);
+
+        $this->assertEquals($expected, $parser->categories());
+    }
+
+    public function testProductsParser()
+    {
+        $path     = __DIR__ . '/fixtures/';
+        $parser   = new Parser($path . 'products.xml');
+        $expected = json_decode(file_get_contents($path . 'products.json'), true);
```
Comment (ziadoz) [inline tests/ParserTest.php]: It might be worth adding a method to `AbstractTest` to do this. I think there's one in the Pangaea SDK you could use.

Comment (weshooper) [inline tests/ParserTest.php]: @ziadoz done, ta!

**Comment by ziadoz** [comment]
:+1:

**Comment by weshooper** [comment]
@ziadoz re-review would be appreciated please - have added more and improved the interface when working with the parser so you can directly request sets/bundles/products/variations only!

File: src/Parser.php [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+        $this->parse($reader);
+    }
+
+    /**
+     * Return an array containing product ids as keys, and an associative array of category ids mapped to whether primary for the values
+     *
+     * @return array
+     */
+    public function assignments()
```
Comment (ziadoz) [inline src/Parser.php]: I think these should probably be `getAssignments()`, `getCategories()` etc. so we have consistent getters and setters.

**Comment by ziadoz** [comment]
Apart from one more minor comment it looks good. :+1:

**Comment by weshooper** [comment]
@ziadoz done ta, plus a final change...

**Comment by ziadoz** [comment]
Looks good. :+1:

**Pull request merged**